src/Tickets/Commerce/Order_Modifiers/Modifiers/Fee.php

```diff
@@ -0,0 +1,292 @@
+		$this->modifier_display_name_plural = __( 'Fees', 'event-tickets' );
+	}
+
+	/**
+	 * Inserts a new modifier and handles related metadata.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data The data to insert.
+	 *
+	 * @return Model The newly inserted modifier or an empty array if no changes were made.
+	 */
+	public function insert_modifier( array $data ): Model {
+		// Save the modifier.
+		$modifier = parent::insert_modifier( $data );
+
+		// Handle metadata (e.g., order_modifier_apply_to).
+		$apply_fee_to = tribe_get_request_var( 'order_modifier_apply_to', '' );
+
+		// Handle metadata (e.g., order_modifier_apply_to).
+		$this->handle_meta_data(
+			$modifier->id,
+			[
+				'meta_key'   => 'fee_applied_to',
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+				'meta_value' => $apply_fee_to,
+			]
+		);
+
+		// Determine the post ID(s) to apply the fee to based on the 'apply_fee_to' value.
+		$apply_to_post_id = null;
+
+		switch ( $apply_fee_to ) {
+			case 'venue':
+				$apply_to_post_id = tribe_get_request_var( 'venue_list', null );
+				break;
+			case 'organizer':
+				$apply_to_post_id = tribe_get_request_var( 'organizer_list', null );
+				break;
+		}
+
+		// Ensure that $apply_to_post_id is an array for consistency.
+		$apply_to_post_ids = $apply_to_post_id ? [ $apply_to_post_id ] : [];
+
+		// Handle the relationship update, passing the relevant data.
+		$this->handle_relationship_update( [ $modifier->id ], $apply_to_post_ids );
+
+		return $modifier;
+	}
+
+	/**
+	 * Updates an existing modifier and handles related metadata.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data The data to update.
+	 *
+	 * @return Model The updated modifier or an empty array if no changes were made.
+	 */
+	public function update_modifier( array $data ): Model {
+		// Save the modifier using the parent method.
+		$modifier = parent::update_modifier( $data );
+
+		// Check if modifier was successfully updated.
+		if ( empty( $modifier ) ) {
+			return [];
+		}
+
+		// Handle metadata (e.g., order_modifier_apply_to).
+		$apply_fee_to = tribe_get_request_var( 'order_modifier_apply_to', '' );
+
+		$this->maybe_clear_relationships( $modifier->id, $apply_fee_to );
+
+		$this->handle_meta_data(
+			$modifier->id,
+			[
+				'meta_key'   => 'fee_applied_to',
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+				'meta_value' => $apply_fee_to,
+			]
+		);
+
+		// Determine the post ID(s) to apply the fee to based on the 'apply_fee_to' value.
+		$apply_to_post_ids = [];
+
+		switch ( $apply_fee_to ) {
+			case 'venue':
+				$apply_to_post_ids = tribe_get_request_var( 'venue_list', [] );
+				break;
+			case 'organizer':
+				$apply_to_post_ids = tribe_get_request_var( 'organizer_list', [] );
+				break;
+		}
+
+		// Ensure that $apply_to_post_id is an array for consistency.
+		$apply_to_post_ids = is_array( $apply_to_post_ids ) ? $apply_to_post_ids : [ $apply_to_post_ids ];
```

Comment:
Why do we do this differently than above?
Cast.

Reply:
Well i wonder the same. Ill cast it in both cases since it makes the most sense. Thank you!